app/db-dump/main.cpp

```diff
@@ -0,0 +1,151 @@
+#include <QtCore/QCoreApplication>
+#include <QtCore/QTimer>
+#include <QtDBus/QtDBus>
+#include <QMessageLogger>
+#include <QtGlobal>
+#include <stdio.h>
+#include <stdlib.h>
+#include "db.h"
+#include "dbinterfaceadaptor.h"
+#include "dbtree.h"
+#include <unistd.h>
+#include <syslog.h>
+#include <comcitrixxenclientdbinterface.h>
+
+typedef struct
+{
+    bool debuggingEnabled;
+    bool syslogEnabled;
+    bool sessionBusEnabled;
+    QString key;
+} CmdLineOptions;
+
+static CmdLineOptions g_cmdLineOptions;
+
+void logOutput(QtMsgType type, const QMessageLogContext&, const QString& msg)
```

Comment:
Probably should put this code into something that is shared between the different apps so that it's not duplicated